### PR TITLE
Bump shellcheck to 0.9.0

### PR DESCRIPTION
**Pull request by thockin**
Fix everything it flagged.

/kind cleanup

```release-note
NONE
```

### CODE REVIEW

File: test/kubemark/run-e2e-tests.sh [hunk 1]
```diff
@@ -39,6 +39,6 @@ fi
 
 # Running locally.
 for ((i=0; i < ${#ARGS[@]}; i++)); do
-  ARGS[$i]="$(echo "${ARGS[$i]}" | sed -e 's/\[/\\\[/g' -e 's/\]/\\\]/g' )"
+  ARGS[i]="$(echo "${ARGS[$i]}" | sed -e 's/\[/\\\[/g' -e 's/\]/\\\]/g' )"
```
Comment (aojea) [inline test/kubemark/run-e2e-tests.sh:42]: I didn't know you could do this

Comment (aojea) [reply test/kubemark/run-e2e-tests.sh:42]: why inside you have to use `$i` and outside no?

**Comment by aojea** [comment]
bash things
/assign @BenTheElder 
/lgtm

Ben is approver in the folder is missing approval

File: test/kubemark/run-e2e-tests.sh (hunk 1 above)
Comment (thockin) [reply test/kubemark/run-e2e-tests.sh:42]: because bash.. <shrug>

**Comment by thockin** [comment]
@BenTheElder  ?

**Comment by pacoxu** [comment]
/hold 
as https://github.com/kubernetes/kubernetes/issues/123127 will fail `pull-kubernetes-node-e2e-containerd` which is required CI.

**Pull request merged**